src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -281,157 +281,66 @@ public void M()
+                //         string? x3 = local(); // warn 3
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 15),
                 // (20,19): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //             string? x4 = null; // warn 10
+                //             string? x4 = null; // warn 5
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(20, 19),
-                // (18,31): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //         string? local(C<string?>? x) // warn 7, 8 and 9
-                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(18, 31),
-                // (18,33): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //         string? local(C<string?>? x) // warn 7, 8 and 9
-                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(18, 33),
                 // (18,15): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //         string? local(C<string?>? x) // warn 7, 8 and 9
+                //         string? local() // warn 4
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(18, 15),
                 // (26,27): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //         System.Func<string?, string?> x5 = (string? x) =>  // warn 12, 13 and 14
+                //         System.Func<string?> x5 = () =>  // warn 6
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(26, 27),
-                // (26,36): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //         System.Func<string?, string?> x5 = (string? x) =>  // warn 12, 13 and 14
-                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(26, 36),
-                // (26,51): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
-                //         System.Func<string?, string?> x5 = (string? x) =>  // warn 12, 13 and 14
-                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(26, 51),
                 // (28,19): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
```

Comment:
>// 4 [](start = 27, length = 4)

No warning on // 4

Reply:
Line 381.

---
In reply to: [208717989](https://github.com/dotnet/roslyn/pull/29127#discussion_r208717989) [](ancestors = 208717989)